pkg/controller/node/node_controller.go

```diff
@@ -170,7 +170,13 @@ func (ctrl *Controller) checkMasterNodesOnAdd(obj interface{}) {
 		glog.V(4).Infof("We don't care about CRs other than cluster created for scheduler config")
 		return
 	}
-	areMastersSchedulable := scheduler.Spec.MastersSchedulable
+	var areMastersSchedulable bool
+	if scheduler.Spec == (configv1.SchedulerSpec{}) {
+		glog.V(4).Infof("Scheduler spec is nil, so set master as unschedulable")
+		areMastersSchedulable = false
```

Comment:
if a bool default value is false, isn't the nil value covered in the original line?  is this change more for the logging or am i missing something?

Reply:
You mean why explicit setting to false if it's already false(the default value for bool is false)? If yes, it's just for my sanity. As the zen of python says `explicit better than implicit`.

Reply:
bool default value in go is false so it seems redundant to do this check unless the logging you are adding has a high value.

Reply:
also the default value of areMastersSchedulable is false so we don't have to reset it there.

Reply:
I generally agree that the extra logging feels...unnecessary but I don't care a lot either.
Take this as a "lgtm" from me, will defer to @kikisdeliveryservice for final lgtm.

Reply:
the more i think about it the more i feel that this should be verified via a test in https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/node/node_controller_test.go or elsewhere.  the check in the code doesn't change any functionality (and imho introduces a bit of confusion at first glance) if you are worried about the treatment of nil values, ensuring they are taken care of via a test makes more sense to me.

Reply:
ie add a test that the existing code fails and if so use this change. if the code does not fail, use as is and keep the test to ensure nothing breaks nil values in the future.